internal/ingress/controller/parser/translate.go

```diff
@@ -373,3 +519,39 @@ func knativeSelectSplit(splits []knative.IngressBackendSplit) knative.IngressBac
 	}
 	return res
 }
+
+func pathsFromK8s(path string, pathType networkingv1.PathType) []*string {
+	switch pathType {
+	case networkingv1.PathTypePrefix:
+		base := strings.Trim(path, "/")
+		if base == "" {
+			return kong.StringSlice("/")
+		}
+		return kong.StringSlice(
+			"/"+base+"$",
+			"/"+base+"/",
+		)
+	case networkingv1.PathTypeExact:
+		relative := strings.TrimLeft(path, "/")
+		return kong.StringSlice("/" + relative + "$")
+	default: // includes ImplementationSpecific
+		if path == "" {
+			return kong.StringSlice("/")
+		}
+		return kong.StringSlice(path)
+	}
+}
+
+// HighestPriority is a Kong regex_priority intended to be higher than any other priority generated by KIC.
+const HighestPriority = 1000000
+
+func priorityForPath(path string, pathType networkingv1.PathType) int {
+	switch pathType {
+	case networkingv1.PathTypeExact:
+		return HighestPriority
+	case networkingv1.PathTypePrefix:
+		return len(path)
+	default:
+		return -HighestPriority
```

Comment:
Can we keep the default as 0?
Users will get confused with a negative priority.

Reply:
This would interfere with the corner case of `Prefix` where the path is an empty string (and thus len 0), so we'd need to shift the `Prefix` priority metric somehow.

I don't have a strong opinion about how we define priorities, so if you have a less confusing scheme, I'm happy to make the change.

Reply:
Invalidated by the solution to the other comment.